src/modules/rlm_perl/rlm_perl.c

```diff
@@ -296,6 +296,40 @@ static XS(XS_radiusd_radlog)
 	XSRETURN_NO;
 }
 
+/*
+ *	This is a wraper for radius_xlat
+ *	Now users are able to get data that is accessible only via xlat
+ *	e.g. %{client:...}
+ *	Call syntax is radiusd::radius_xlat(string), string will be handled the
+ *	same way it is described in EXPANSIONS section of man unlang
+ */
+static XS(XS_radiusd_radius_xlat)
+{
+	dXSARGS;
+	char *in_str;
+	char expanded[1024]; /* XXX FIXME Dynamic allocation in case slen >= sizeof(expanded) ? */
+	ssize_t slen;
+	SV *rad_requestp_sv;
+	REQUEST *request;
+
+	if (items != 1) croak("Usage: radiusd::radius_xlat(string)");
+
+	rad_requestp_sv = get_sv("RAD___REQUESTP", 0);
```

Comment:
Do it like this instead:
- https://github.com/FreeRADIUS/freeradius-server/blob/v4.0.x/src/modules/rlm_lua/lua.c#L40
- https://github.com/FreeRADIUS/freeradius-server/blob/v4.0.x/src/modules/rlm_lua/lua.c#L247
- https://github.com/FreeRADIUS/freeradius-server/blob/v4.0.x/src/modules/rlm_lua/lua.c#L770

Many many many fewer cycles.